src/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    internal ref struct ValueStringBuilder
+    {
+        private char[] _arrayToReturnToPool;
```

Comment:
it would be more readable if instead of char[] and Span<char> you just had Span<char>  plus a boolean indicating whether it wrapped an array from the pool. But I guess there is no reasonable way to get the array out of the Span<char>? (Perhaps with some Unsafe magic)

Reply:
> it would be more readable if instead of char[] and Span you just had Span plus a boolean indicating whether it wrapped an array from the pool

You can't get the array out of the span.  And any efforts to do so would be way less readable ;)